std/haxe/macro/ExprTools.hx

```diff
@@ -128,6 +128,11 @@ class ExprTools {
 				}
 				if (edef != null && edef.expr != null)
 					f(edef);
+			case EFormat(parts):
+				for (part in parts) switch part.kind {
+					case FRaw(_) | FIdent(_):
```

Comment:
Why not create an adhoc expression and deal with it accordingly?

iter: 
```haxe
case FIdent(name): f(macro @:pos(part.pos) $i{name})
```

map:
```haxe
case FIdent(name): {
  pos: part.pos,
  kind: switch f(macro @:pos(part.pos) $i{name}) {
     case macro $i{ident}: FIdent(name);
     case v: FExpr(v);
  }
}
```